generator/maxspeed_collector.cpp

```diff
@@ -0,0 +1,80 @@
+#include "generator/maxspeed_collector.hpp"
+
+#include "coding/file_writer.hpp"
+
+#include "base/geo_object_id.hpp"
+#include "base/logging.hpp"
+
+#include <fstream>
+#include <sstream>
+
+namespace feature
+{
+using namespace base;
+using namespace std;
+
+void MaxspeedCollector::Process(OsmElement const & p)
+{
+  ostringstream ss;
+  ss << p.id << ",";
+
+  auto const & tags = p.Tags();
+  string maxspeedForward;
+  string maxspeedBackward;
+  bool isReverse = false;
+
+  for (auto const & t : tags)
+  {
+    if (t.key == "maxspeed")
+    {
+      ss << t.value;
+      m_data.push_back(ss.str());
+      return;
+    }
+
+    if (t.key == "maxspeed:forward")
+      maxspeedForward = t.value;
+    else if (t.key == "maxspeed:backward")
+      maxspeedBackward = t.value;
+    else if (t.key == "oneway")
+      isReverse = (t.value == "-1");
+  }
+
+  // Note 1. isReverse == true means feature |p| has tag "oneway" with value "-1". Now (10.2018)
+  // no feature with a tag oneway==-1 and a tag maxspeed:forward/backward is found. But to
+  // be on the safe side the case is processed.
```

Comment:
s/processed/handled/
@mpimenov ?

Reply:
Yes, handled is better.
s/the case/this case/

Reply:
Ok. handled, this case.